wasm/src/value.c

```diff
@@ -1521,6 +1560,19 @@ int _validate_json_path(opa_value *path)
     return path_len;
 }
 
+OPA_INTERNAL
+opa_value *opa_string_copy(opa_string_t *src)
```

Comment:
So we need to copy the keys now on `opa_value_add_path` because the `*v` will eventually be free'ed by the caller. Correct?

Reply:
Correct.  Unfortuantely, opa_value_add_path() (more specifically __opa_object_insert()) may or may not find that a given object already has a particular path element string as a key already.  If it doesn't, the data doc needs to keep a copy of the string.  If it does, we need to make sure to release one of them.   In https://github.com/open-policy-agent/opa/issues/5785 I discussed this in the section **Another Leak: opa_object_insert()**.  

In point of fact, there are other functions that make shallow copies of objects or create set unions, etc, that may leak some random key strings as a result of the inconsistent key handling.  But, AFAICT, those functions are always called during eval() operations and so get freed implicitly at the next heap pointer reset.

In this patch, it's only opa_value_add_path() that invokes the __opa_object_insert() with a copy of the key.  If the key already exists, it frees the fresh copy (leaves the original in place).  This means it was wasted CPU to copy in the first place.  But
1. this isn't going to be an operation we perform at the frequency of queries and
2. it makes for a cleaner caller API than forcing the caller to track whether to free the key via a return code or some such